samples/snippets/src/main/java/com/example/monitoring/UpdateAlertPolicy.java

```diff
@@ -0,0 +1,98 @@
+public class UpdateAlertPolicy {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String alertPolicyName = "your-alert-policy-displayname";
+    String updatedAlertPolicyName = "your-updated-alert-policy-displayname";
+    updateAlertPolicy(alertPolicyName, updatedAlertPolicyName);
+  }
+
+  public static void updateAlertPolicy(String alertPolicyName, String newPolicyName)
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (AlertPolicyServiceClient alertPolicyServiceClient = AlertPolicyServiceClient.create()) {
+      // A Filter that identifies which time series should be compared with the threshold
+      String metricFilter =
+          "metric.type=\"compute.googleapis.com/instance/"
+              + "cpu/utilization\" AND resource.type=\"gce_instance\"";
+
+      // Build Duration
+      Duration aggregationDuration = Duration.newBuilder().setSeconds(60).build();
+
+      // Build Aggregation
+      Aggregation aggregation =
+          Aggregation.newBuilder()
+              .setAlignmentPeriod(aggregationDuration)
+              .setCrossSeriesReducer(Aggregation.Reducer.REDUCE_MEAN)
+              .setPerSeriesAligner(Aggregation.Aligner.ALIGN_MAX)
+              .build();
+
+      // Build MetricThreshold
+      AlertPolicy.Condition.MetricThreshold metricThreshold =
+          AlertPolicy.Condition.MetricThreshold.newBuilder()
+              .setComparison(ComparisonType.COMPARISON_GT)
+              .addAggregations(aggregation)
+              .setFilter(metricFilter)
+              .setDuration(aggregationDuration)
+              .build();
+
+      // build Alert policy condition
+      AlertPolicy.Condition alertPolicyCondition =
+          AlertPolicy.Condition.newBuilder()
+              .setDisplayName(alertPolicyName)
+              .setConditionThreshold(metricThreshold)
+              .build();
+
+      // create alert policy for update
+      AlertPolicy alertPolicy =
+          AlertPolicy.newBuilder()
+              .setName(alertPolicyName)
+              .setDisplayName(newPolicyName)
+              .addConditions(alertPolicyCondition)
+              .setCombiner(AlertPolicy.ConditionCombinerType.AND)
+              .build();
+
+      // create update alert policy request
+      UpdateAlertPolicyRequest updateAlertPolicyRequest =
+          UpdateAlertPolicyRequest.newBuilder().setAlertPolicy(alertPolicy).build();
+
+      // process update policy request
+      AlertPolicy updatedAlertPolicy =
+          alertPolicyServiceClient.updateAlertPolicy(updateAlertPolicyRequest);
+      System.out.print("alert policy updated successfully:" + updatedAlertPolicy.getDisplayName());
+    } catch (ApiException ex) {
+      System.out.print("alert policy not found:" + ex.toString());
+    }
```

Comment:
If you aren't doing anything with an exception, you probably shouldn't be catching it.  
See our [sample format guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#exception-handling) for more info.

Reply:
Done